src/fastaq_handler.cpp

```diff
@@ -105,6 +118,7 @@ void FastaqHandler::get_id(const uint32_t& id)
         read.clear();
         gzrewind(this->fastaq_file);
         kseq_rewind(this->inbuf);
+        this->read_ahead_next_read();
```

Comment:
When we rewind the file, we have to prepare the first sequence to be read. **Note: the annoying thing of this read ahead implementation is that the next sequence to be read always have to be prepared (i.e. read ahead) before the user request it. I think the implementation is correct, but it would be nice if you could double check (maybe with test cases) that `FastaqHandler::skip_next()` and `FastaqHandler::get_id()` is doing the correct job of preparing the next read.**